src/mlpack/methods/lsh/lsh_search_impl.hpp

```diff
@@ -339,12 +340,301 @@ void LSHSearch<SortPolicy>::BaseCase(const size_t queryIndex,
         referenceIndex, distance);
 }
 
+
+// Compare class for <double, size_t> pair, used in GetAdditionalProbingBins.
+class CompareGreater
+{
+  public:
+    bool operator()(
+        std::pair<double, size_t> p1,
+        std::pair<double, size_t> p2){
+      //only compare the double values
+      return p1.first > p2.first;
+    }
```

Comment:
Are you sure `std::greater` will not work here instead of this class?  By default for `std::pair` that will take into account the second element, but I think that in your case ordering based on the second element doesn't really matter.